From bc20961b4e6eb8f4124a7e0c0af81691644e3bf9 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 11 Jan 2019 16:50:05 +0200
Subject: [PATCH 493/706] enetc: set MAC IF_MODE for SXGMII per BG

depends on boot loader information passed through IERB.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 5d2e924d42fe40d53e819f544bc824240c9c5663)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 4 ++++
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index d7c31c1a019b..048601be660b 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -217,6 +217,7 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_PMO_IFM_RG	BIT(2)
 #define ENETC_PM0_IFM_RLP	(BIT(5) | BIT(11))
 #define ENETC_PM0_IFM_RGAUTO	(BIT(15) | ENETC_PMO_IFM_RG | BIT(1))
+#define ENETC_PM0_IFM_XGMII	BIT(12)
 
 /* MAC counters */
 #define ENETC_PM0_REOCT		0x8100
@@ -281,6 +282,9 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_GLOBAL_BASE	0x20000
 #define ENETC_G_EIPBRR0		0x0bf8
 #define ENETC_G_EIPBRR1		0x0bfc
+#define ENETC_G_EPFBLPR(n)	(0xd00 + 4 * (n))
+#define ENETC_G_EPFBLPR1_XGMII	0x80000000
+
 
 /* PCI device info */
 struct enetc_hw {
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 13fcfeca197b..bf934a8dd80e 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -525,6 +525,8 @@ static void enetc_configure_port_mac(struct enetc_hw *hw)
 	/* set auto-speed for RGMII */
 	if (enetc_port_rd(hw, ENETC_PM0_IF_MODE) & ENETC_PMO_IFM_RG)
 		enetc_port_wr(hw, ENETC_PM0_IF_MODE, ENETC_PM0_IFM_RGAUTO);
+	if (enetc_global_rd(hw, ENETC_G_EPFBLPR(1)) == ENETC_G_EPFBLPR1_XGMII)
+		enetc_port_wr(hw, ENETC_PM0_IF_MODE, ENETC_PM0_IFM_XGMII);
 }
 
 static void enetc_configure_port_pmac(struct enetc_hw *hw)
-- 
2.17.1

